test/integration/test_convention_architecture.py

```diff
@@ -0,0 +1,95 @@
+"""
+Tests for the new convention architecture.
+"""
+import numpy as np
+import xarray as xr
+from affine import Affine
+
+import rioxarray
+from rioxarray.enum import Convention
+
+
+class TestConventionArchitecture:
+    """Test integration scenarios for the convention architecture."""
+
+    def test_set_options_convention(self):
+        """Test setting convention through set_options."""
+        # Test default convention (None for CF-first with Zarr fallback)
+        with rioxarray.set_options():
+            from rioxarray._options import CONVENTION, get_option
+
+            assert get_option(CONVENTION) is None
+
+        # Test setting Zarr convention
+        with rioxarray.set_options(convention=Convention.Zarr):
+            from rioxarray._options import CONVENTION, get_option
```

Comment:
Is there a reason the imports are here instead of at the top?

Reply:
This appears to be a common pattern. If possible, would be great for all imports to be at the top of the file.